UI/MapWnd.cpp

```diff
@@ -4119,24 +4118,26 @@ void MapWnd::InitFieldRenderingBuffers() {
     }
     m_field_scanline_circles.createServerBuffer();
 
-    for (auto& [field_texture, buffers] : m_field_vertices) {
-        if (!field_texture)
-            continue;
-
-        // TODO: why the binding here?
-        glBindTexture(GL_TEXTURE_2D, field_texture->OpenGLId());
-        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP);
-        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP);
+    std::size_t max_buffer_size = 0;
 
-        buffers.first.createServerBuffer();
-        buffers.second.createServerBuffer();
+    auto field_texture_exists = [](const auto& field_vertices_pair) { return !!field_vertices_pair.first; };
```

Comment:
bit odd to write `!!` rather than make the lambda return bool with `-> bool`

Reply:
I seemed to recall that "back in my day" it was considered safer to do double bang than relying on operator bool. Tried to find something on it, search came with https://stackoverflow.com/questions/3193201/how-does-shared-ptr-safely-allow-casting-to-bool and talks about safe bool idiom and how such stuff may no longer be necessary in post C++11 world. Well shared_ptr seems to be in std since then, too... Before you wrote your own stuff or had boost's shared_ptr (which probably was doing the safe bool idiom anyway so perhaps was okay to relay on operator bool, too, in boost's case?) 

Might be some ancient superstition, doesn't look that weird to me, but I'll be hip and down with the cool kids or something,, hah!! :) I'll change it!!

Reply:
whelp, I tried `->bool` and it complains 

Error (active)	E0413	no suitable conversion function from "const std::shared_ptr<GG::Texture>" to "bool" exists	

I can however explicitly use `operator bool` though personally would rather do double bang indeed. Sigh. Will do `static constexpr auto field_texture_exists = [](const auto& field_vertices_pair) { return (bool)field_vertices_pair.first; };`

Reply:
btw that check seems too defensive to me anyway, is there really a chance the texture would be `nullptr`? but the original code had it so I kept it

Reply:
another option is probably `return field_vertices_pair.first.get();`

Reply:
OK I'll do get(), that's what the std library implementation for MSVC does anyway under the hood for `(bool)`

Reply:
(though it compares to `nullptr` rather than doing pointer casting to bool or whatever)

Reply:
should be able to just return whatever.get() which should be implicitly casted to bool if that's the return type of the lambda. the problem without .get() is that the operator bool() of shared_ptr is explicit, but raw pointer casts to bool are automatic and the result is the same as != nullptr (as far as I know).